distributed/client.py

```diff
@@ -858,8 +858,9 @@ def __init__(
     ):
         if timeout is no_default:
             timeout = dask.config.get("distributed.comm.timeouts.connect")
-        if timeout is not None:
-            timeout = parse_timedelta(timeout, "s")
+        timeout = parse_timedelta(timeout, "s")
+        if timeout is None:
+            raise ValueError("None is an invalid value for global client timeout")
```

Comment:
There are so many places in the client code that just blindly assume it's not None. The json schema does not allow for None either.